c/zss.c

```diff
@@ -317,10 +319,38 @@ static void setPrivilegedServerName(HttpServer *server, JsonObject *mvdSettings,
 }
 #endif /* __ZOWE_OS_ZOS */
 
+static int nativeWithSessionTokenAuth(HttpService *service, HttpRequest *request, HttpResponse *response) {
+  int rc = 0;
+  char profileName[ZOWE_PROFILE_NAME_LEN+1] = {0};
+  char method[16];
+  snprintf(method, sizeof(method), "%s", request->method);
+  destructivelyNativize(method);
+  rc = getProfileNameFromRequest(profileName, request->parsedFile, method, -1, response);
+  if (rc != 0) {
+    return FALSE;
+  }
+  rc = serveAuthCheckByParams(service, request->username, "ZOWE", profileName, 2, ENV_SETTINGS);
+  if (rc != 0) {
+    return FALSE;
+  }
+  return TRUE;
+}
+
+/* Future custom ZSS authorization handlers go here */
+static void initializeAuthHandlers(HttpServer *server) {
+  
+  /* NATIVE_WITH_SESSION_TOKEN */ 
+  server->authHandler[0] = (HttpAuthHandler*)safeMalloc31(sizeof(HttpAuthHandler),"HttpAuthHandler");
+  server->authHandler[0]->type = SERVICE_AUTH_NATIVE_WITH_SESSION_TOKEN;
+  server->authHandler[0]->authFunction = &nativeWithSessionTokenAuth;
```

Comment:
It's better not to change the internals of the HTTP server by hand, you can expose this functionality via a public function in `zowe-common-c/h/httpserver.h`.

Maybe something like this?
```c
int httpServerAddAuthHandler(HttpServer *self, int type, AuthHandle *handler);
```